libraries/chain/include/eosio/chain/block_state.hpp

```diff
@@ -20,46 +20,62 @@ inline weak_digest_t create_weak_digest(const digest_type& digest) {
 }
 
 struct block_state_legacy;
+struct block_state_accessor;
 
 struct block_state : public block_header_state {     // block_header_state provides parent link
    // ------ data members -------------------------------------------------------------
    signed_block_ptr           block;
-   bool                       validated = false;     // We have executed the block's trxs and verified that action merkle root (block id) matches.
    digest_type                strong_digest;         // finalizer_digest (strong, cached so we can quickly validate votes)
    weak_digest_t              weak_digest;           // finalizer_digest (weak, cached so we can quickly validate votes)
    pending_quorum_certificate pending_qc;            // where we accumulate votes we receive
    std::optional<valid_quorum_certificate> valid_qc; // best qc received from the network inside block extension
 
+   // ------ updated for votes, used for fork_db ordering ------------------------------
+private:
+   bool                       validated = false;            // We have executed the block's trxs and verified that action merkle root (block id) matches.
+   qc_claim_t                 most_recent_ancestor_with_qc; // only modify/access while holding forkdb lock
```

Comment:
I think the name `best_qc_claim` would be better. I was not sure we necessarily want to update to a higher block_num with a weak qc if the one we have is is strong.

Reply:
I like `best_qc_claim` as well. I was going with what @arhag had in #2125. If no objection from @arhag I'll change it to `best_qc_claim`.